sdk/cosmos/azure-cosmos/azure/cosmos/auth.py

```diff
@@ -27,12 +27,23 @@
 import hmac
 
 import six
+import warnings
 
 from . import http_constants
 
 
 def GetAuthorizationHeader(
-    cosmos_client_connection, verb, path, resource_id_or_fullname, is_name_based, resource_type, headers
+        cosmos_client_connection, verb, path, resource_id_or_fullname, is_name_based, resource_type, headers
+):
+    warnings.warn("This method is being removed from the SDK, please use recommended authorization patterns by passing "
+                  "in credentials directly within the creation of the cosmos client instance", UserWarning)
```

Comment:
I think we should use the `DeprecationWarning` flag here rather than `UserWarning`.
I would also phrase the message accordingly - something like:
"This method has been deprecated and will be removed in a future release."